### PR TITLE
feat(argo-cd): add lifecycle hooks to repo server

### PR DESCRIPTION
**Pull request by gosharo**
Signed-off-by: goshado <goshatoo@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ appVersion: v2.6.7
 kubeVersion: ">=1.22.0-0"
 description: A Helm chart for Argo CD, a declarative, GitOps continuous delivery tool for Kubernetes.
 name: argo-cd
-version: 5.28.1
+version: 5.28.2
```
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:6]: Please also update the `artifacthub.io/changes: |` some lines below (L26 + L27)

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

File: charts/argo-cd/Chart.yaml (hunk 1 above)
Comment (gosharo) [reply charts/argo-cd/Chart.yaml]: @mkilchhofer done

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Please regenerate the readme with the correct helm-docs version. You can use the `scripts/helm-docs.sh` script for that.

Oh and please bump the chart version (minor version)

File: charts/argo-cd/Chart.yaml [hunk 2]
```diff
@@ -26,5 +26,5 @@ annotations:
     fingerprint: 2B8F22F57260EFA67BE1C5824B11F800CD9D2252
     url: https://argoproj.github.io/argo-helm/pgp_keys.asc
   artifacthub.io/changes: |
-    - kind: changed
-      description: Update redis-ha to v4.23.0
+    - kind: added
+      description: Add lifesycle webhook to repo server.
```
Comment (tico24) [inline charts/argo-cd/Chart.yaml:30]: nit: typo

**Review by tico24 — CHANGES_REQUESTED** [review]

File: charts/argo-cd/Chart.yaml (hunk 2 above)
Comment (gosharo) [reply charts/argo-cd/Chart.yaml]: thanks

**Review by tico24 — APPROVED** [review]

**Review by yu-croco — APPROVED** [review]
LGTM

**Review by jmeridth — APPROVED** [review]

**Review by mbevc1 — APPROVED** [review]

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**